src/config-import.js

```diff
@@ -0,0 +1,271 @@
+     */
+    *fields() {
+      const { fields } = presetsFile
+      for (const [name, field] of Object.entries(fields)) {
+        if (!isRecord(field) && !hasOwn(field, 'key')) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        /** @type {Record<string,unknown>} */
+        const fieldValue = {
+          schemaName: 'field',
+          tagKey: field.key,
+        }
+        for (const key of Object.keys(valueSchemas.field.properties)) {
+          if (key in field) {
+            // @ts-ignore - we validate below
+            fieldValue[key] = field[key]
+          }
+        }
+        if (!validate('field', fieldValue)) {
+          warnings.push(new Error(`Invalid field ${name}`))
+          continue
+        }
+        yield {
+          name,
+          value: fieldValue,
+        }
+      }
+    },
+
+    /**
+     * @returns {Iterable<{ fieldNames: string[], iconName: string | undefined, value: import('@mapeo/schema').PresetValue }>}
+     */
+    *presets() {
+      const { presets } = presetsFile
+      // sort presets using the sort field, turn them into an array
+      const sortedPresets = Object.keys(presets)
+        .filter((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          const isInvalidPreset = !isRecord(preset)
+          if (isInvalidPreset) {
+            warnings.push(new Error(`invalid preset ${presetName}`))
+          }
+          return !isInvalidPreset
+        })
+        .map((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          if (!preset.sort) {
+            // if there's no sort field, put a big value - puts the field at the end -
+            preset.sort = 100
+          }
+          return preset
+        })
+        .sort((preset, nextPreset) => nextPreset.sort - preset.sort)
+
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let preset of sortedPresets) {
+        /** @type {any} */
+        const presetValue = {
+          schemaName: 'preset',
+          fieldIds: [],
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
+            presetValue[key] = preset[key]
+          }
+        }
+        if (!validate('preset', presetValue)) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        yield {
+          fieldNames:
+            'fields' in preset && Array.isArray(preset.fields)
+              ? preset.fields
+              : [],
+          iconName:
+            'icon' in preset && typeof preset.icon === 'string'
+              ? preset.icon
+              : undefined,
+          value: presetValue,
+        }
+      }
+    },
```

Comment:
I'd love if we could remove some of the `any`s and `ts-ignore`s in this function. Here's my take on how to do that:

```suggestion
    *presets() {
      const { presets } = presetsFile

      /** @type {Record<string, unknown>[]} */
      const validPresets = []
      for (const [presetName, preset] of Object.entries(presets)) {
        if (isRecord(preset)) {
          validPresets.push(preset)
        } else {
          warnings.push(new Error(`invalid preset ${presetName}`))
        }
      }

      validPresets.sort((a, b) => {
        const aSort = typeof a === 'number' ? a : Infinity
        const bSort = typeof b === 'number' ? b : Infinity
        return bSort - aSort
      })

      for (const preset of validPresets) {
        /** @type {Record<string, unknown>} */
        const presetValue = {
          schemaName: 'preset',
          fieldIds: [],
          addTags: {},
          removeTags: {},
          terms: [],
        }
        for (const key of Object.keys(valueSchemas.preset.properties)) {
          if (hasOwn(preset, key)) {
            presetValue[key] = preset[key]
          }
        }
        if (!validate('preset', presetValue)) {
          warnings.push(new Error(`Invalid preset ${preset.name}`))
          continue
        }
        yield {
          fieldNames:
            'fields' in preset &&
            Array.isArray(preset.fields) &&
            preset.fields.every((f) => typeof f === 'string')
              ? preset.fields
              : [],
          iconName:
            'icon' in preset && typeof preset.icon === 'string'
              ? preset.icon
              : undefined,
          value: presetValue,
        }
      }
    },
```

(Pretty similar to what you had, just removing some `any`s.)